src/metabase/api/notify.clj

```diff
@@ -4,19 +4,29 @@
             [metabase.api.common :as api]
             [metabase.models.database :refer [Database]]
             [metabase.models.table :refer [Table]]
-            [metabase.sync :as sync]))
+            [metabase.sync :as sync]
+            [metabase.sync.sync-metadata :as sync-metadata]))
 
 (api/defendpoint POST "/db/:id"
   "Notification about a potential schema change to one of our `Databases`.
-  Caller can optionally specify a `:table_id` or `:table_name` in the body to limit updates to a single `Table`."
-  [id :as {{:keys [table_id table_name]} :body}]
-  (api/let-404 [database (Database id)]
-    (cond
-      table_id (when-let [table (Table :db_id id, :id (int table_id))]
-                 (future (sync/sync-table! table)))
-      table_name (when-let [table (Table :db_id id, :name table_name)]
-                   (future (sync/sync-table! table)))
-      :else (future (sync/sync-database! database))))
+  Caller can optionally specify a `:table_id` or `:table_name` in the body to limit updates to a single
+  `Table`. Optional Parameter `:scan` can be `\"full\" or \"schema\" for a full sync or a schema sync, available
+  regardless if a `:table_id` or `:table_name` is passed."
+  [id :as {{:keys [table_id table_name scan]} :body}]
+  (when scan
+    (or (contains? #{"full" :full "schema" :schema} scan)
+        (throw (ex-info "Optional scan parameter must be either \"full\" or \"scan\""
+                        {:status-code 400}))))
```

Comment:
params are never going to be passed in to the API as keywords, so you don't need to check that.

We have a pattern to check API endpoint parameters using Schema. `defendpoint` will convert a schema map directly after the arglist into a series of appropriate checks. On top of that, it will include those constraints in the API endpoint docstring. Yours should look something like:

```clj
{scan (s/optional (s/enum "full "schema"))}
```